The job **`aarch64-gnu-llvm-20-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/19876736315/job/56965906478) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/56965906478) [(plain)](https://github.com/rust-lang/rust/commit/dd65f8b383c058aee836fa1662aa276f88dd864b/checks/56965906478/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

failures:

---- [ui] tests/ui/feature-gates/feature-gate-cfg-emscripten-wasm-eh.rs stdout ----
Saved the actual stderr to `/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/feature-gates/feature-gate-cfg-emscripten-wasm-eh/feature-gate-cfg-emscripten-wasm-eh.stderr`
diff of stderr:

1 error[E0658]: `cfg(emscripten_wasm_eh)` is experimental and subject to change
-   --> $DIR/feature-gate-cfg-emscripten-wasm-eh.rs:2:11
+   --> $DIR/feature-gate-cfg-emscripten-wasm-eh.rs:2:7
3    |
- LL | #[cfg(not(emscripten_wasm_eh))]
-    |           ^^^^^^^^^^^^^^^^^^
+ LL | #[cfg(emscripten_wasm_eh)]
+    |       ^^^^^^^^^^^^^^^^^^
6    |
7    = help: add `#![feature(cfg_emscripten_wasm_eh)]` to the crate attributes to enable
8    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

Note: some mismatched output was normalized before being compared
-   --> /checkout/tests/ui/feature-gates/feature-gate-cfg-emscripten-wasm-eh.rs:2:7
+   --> $DIR/feature-gate-cfg-emscripten-wasm-eh.rs:2:7
+ LL | #[cfg(emscripten_wasm_eh)]
+    |       ^^^^^^^^^^^^^^^^^^


The actual stderr differed from the expected stderr
To update references, rerun the tests and pass the `--bless` flag
To only update this specific test, also pass `--test-args feature-gates/feature-gate-cfg-emscripten-wasm-eh.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/feature-gates/feature-gate-cfg-emscripten-wasm-eh.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/feature-gates/feature-gate-cfg-emscripten-wasm-eh" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "--check-cfg=cfg(emscripten_wasm_eh)"
stdout: none
--- stderr -------------------------------
error[E0658]: `cfg(emscripten_wasm_eh)` is experimental and subject to change
##[error]  --> /checkout/tests/ui/feature-gates/feature-gate-cfg-emscripten-wasm-eh.rs:2:7
   |
LL | #[cfg(emscripten_wasm_eh)]
   |       ^^^^^^^^^^^^^^^^^^
   |
   = help: add `#![feature(cfg_emscripten_wasm_eh)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0658`.
------------------------------------------
```

</details>

--- a/compiler/rustc_interface/src/tests.rs
+++ b/compiler/rustc_interface/src/tests.rs
@@ -798,7 +798,7 @@ fn test_unstable_options_tracking_hash() {
     tracked!(embed_metadata, false);
     tracked!(embed_source, true);
     tracked!(emit_thin_lto, false);
-    tracked!(emscripten_wasm_eh, true);
+    tracked!(emscripten_wasm_eh, false);
     tracked!(export_executable_symbols, true);
     tracked!(fewer_names, Some(true));
     tracked!(fixed_x18, true);

--- a/compiler/rustc_session/src/options.rs
+++ b/compiler/rustc_session/src/options.rs
@@ -2338,7 +2338,7 @@ options! {
         "emit a section containing stack size metadata (default: no)"),
     emit_thin_lto: bool = (true, parse_bool, [TRACKED],
         "emit the bc module with thin LTO info (default: yes)"),
-    emscripten_wasm_eh: bool = (false, parse_bool, [TRACKED],
+    emscripten_wasm_eh: bool = (true, parse_bool, [TRACKED],
         "Use WebAssembly error handling for wasm32-unknown-emscripten"),
     enforce_type_length_limit: bool = (false, parse_bool, [TRACKED],
         "enforce the type length limit when monomorphizing instances in codegen"),

--- a/src/doc/rustc/src/platform-support/wasm32-unknown-emscripten.md
+++ b/src/doc/rustc/src/platform-support/wasm32-unknown-emscripten.md
@@ -95,10 +95,11 @@ This target can be cross-compiled from any host.
 ## Emscripten ABI Compatibility
 
 The Emscripten compiler toolchain does not follow a semantic versioning scheme
-that clearly indicates when breaking changes to the ABI can be made. Additionally,
-Emscripten offers many different ABIs even for a single version of Emscripten
-depending on the linker flags used, e.g. `-fexceptions` and `-sWASM_BIGINT`. If
-the ABIs mismatch, your code may exhibit undefined behaviour.
+that clearly indicates when breaking changes to the ABI can be made.
+Additionally, Emscripten offers many different ABIs even for a single version of
+Emscripten depending on the linker flags used, e.g. `-fwasm-exceptions` and
+`-sWASM_BIGINT`. If the ABIs do not match, your code may exhibit undefined
+behaviour.
 
 To ensure that the ABIs of your Rust code, of the Rust standard library, and of
 other code compiled for Emscripten all match, you should rebuild the Rust standard
@@ -158,9 +159,9 @@ features can be disabled, and how Rust code can be conditionally compiled based 
 which features are enabled.
 
 Note that Rust code compiled for `wasm32-unknown-emscripten` currently enables
-`-fexceptions` (JS exceptions) by default unless the Rust code is compiled with
-`-Cpanic=abort`. `-fwasm-exceptions` (WASM exceptions) is not yet currently supported,
-see <https://github.com/rust-lang/rust/issues/112195>.
+`-fwasm-exceptions` (legacy WASM exceptions) by default unless the Rust code is
+compiled with `-Cpanic=abort`. It is possible to use JS exceptions by passing
+the flag ``-Z emscripten-wasm-eh=false`` but this will be removed in the future.
 
 Please refer to the [Emscripten ABI compatibility](#emscripten-abi-compatibility)
 section to ensure that the features that are enabled do not cause an ABI mismatch

--- a/tests/codegen-llvm/emscripten-catch-unwind-js-eh.rs
+++ b/tests/codegen-llvm/emscripten-catch-unwind-js-eh.rs
@@ -1,4 +1,4 @@
-//@ compile-flags: -Copt-level=3 --target wasm32-unknown-emscripten
+//@ compile-flags: -Copt-level=3 --target wasm32-unknown-emscripten -Z emscripten-wasm-eh=false
 //@ needs-llvm-components: webassembly
 
 // Emscripten has its own unique implementation of catch_unwind (in `codegen_emcc_try`),

--- a/tests/ui/feature-gates/feature-gate-cfg-emscripten-wasm-eh.rs
+++ b/tests/ui/feature-gates/feature-gate-cfg-emscripten-wasm-eh.rs
@@ -1,4 +1,4 @@
 //@ compile-flags: --check-cfg=cfg(emscripten_wasm_eh)
-#[cfg(not(emscripten_wasm_eh))]
+#[cfg(emscripten_wasm_eh)]
 //~^ ERROR `cfg(emscripten_wasm_eh)` is experimental
 fn main() {}

--- a/tests/ui/feature-gates/feature-gate-cfg-emscripten-wasm-eh.stderr
+++ b/tests/ui/feature-gates/feature-gate-cfg-emscripten-wasm-eh.stderr
@@ -1,8 +1,8 @@
 error[E0658]: `cfg(emscripten_wasm_eh)` is experimental and subject to change
-  --> $DIR/feature-gate-cfg-emscripten-wasm-eh.rs:2:11
+  --> $DIR/feature-gate-cfg-emscripten-wasm-eh.rs:2:7
    |
-LL | #[cfg(not(emscripten_wasm_eh))]
-   |           ^^^^^^^^^^^^^^^^^^
+LL | #[cfg(emscripten_wasm_eh)]
+   |       ^^^^^^^^^^^^^^^^^^
    |
    = help: add `#![feature(cfg_emscripten_wasm_eh)]` to the crate attributes to enable
    = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

